### PR TITLE
Add support for extraction of wordings in form type classes

### LINKED ISSUE #45: Extract choice type translations
In the following file: https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/ProductPreferences/PaginationType.php

There are two new extraction cases:

1. [Choices using an explicit translation domain](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/ProductPreferences/PaginationType.php#L61-L68)

2. [Choices not including an implicit translation domain](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/ProductPreferences/PaginationType.php#L48-L60), where a [default translation domain is defined](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/ProductPreferences/PaginationType.php#L74-L79)

### PR DESCRIPTION
**Pull request by eternoendless**
This PR includes #48 and Fixes #45 

This allows to extract wordings declared in ChoiceType form elements.
For the extractor to work, a specific criteria must be met:

1. The class must extend AbstractType or TranslatorAwareType
2. Wordings must be added using this specific syntax:
	```php
	->add('some_name', ChoiceType::class, [
		// [...]
		'choices' => [
			// choices go here
		]
	])
	```
3. The translation domain must be declared as the `choice_translation_domain` property of the field, or declared as default domain.

If the translation domain is declared as default, then its declaration must meet the following criteria:

1. It must be declared inside the `configureOptions()` method.
2. It must be inside `$resolver->setDefaults()`. The variable `$resolver` can be named in any way as long as it's the same as the first parameter of `configureOptions()`.

### IMPORTANT NOTICE

The code is analyzed statically. If you use variable interpolation, the extraction WILL NOT WORK.

Some examples that won't work:
```php
public function configureOptions(OptionsResolver $resolver)
{
	$defaultDomain = 'Admin.Shopparameters.Feature';
	$resolver->setDefaults([
		'translation_domain' => $defaultDomain, // variable is not interpolated
	]);
}
```

```php
public function configureOptions(OptionsResolver $resolver)
{
	$renamedResolver = $resolver; // not supported
	$renamedResolver->setDefaults([
		'translation_domain' => 'Admin.Shopparameters.Feature',
	]);
}
```

### CODE REVIEW

File: Translation/Extractor/PhpExtractor.php [hunk 1]
```diff
@@ -97,21 +108,47 @@ protected function parseFileTokens($file, MessageCatalogue $catalog)
...
+        $traverser->addVisitor($commentsNodeVisitor);
+        foreach ($translationVisitors as $visitor) {
+            $traverser->addVisitor($visitor);
+        }
+
+        $nodeDumper = new NodeDumper();
 
         try {
             $stmts = $this->parser->parse($code);
+            //$debug = $nodeDumper->dump($stmts);
```
Comment (PierreRambaud) [inline Translation/Extractor/PhpExtractor.php]: Keep debug?

File: Translation/Extractor/Util/TranslationCollection.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<?php
```
Comment (PierreRambaud) [inline Translation/Extractor/Util/TranslationCollection.php]: Missing license

Comment (eternoendless) [reply Translation/Extractor/Util/TranslationCollection.php]: This repo has no license yet!

Comment (PierreRambaud) [reply Translation/Extractor/Util/TranslationCollection.php]: Weird when we have license block in many files :(

Comment (eternoendless) [reply Translation/Extractor/Util/TranslationCollection.php]: That is clearly a mistake. We only release the core in OSL.

**Pull request merged**